python_modules/libraries/dagster-looker/dagster_looker/dagster_looker_translator.py

```diff
@@ -1,25 +1,168 @@
+import itertools
+import logging
+import re
 from pathlib import Path
-from typing import Any, Mapping, Optional, Sequence, Tuple
+from typing import Any, Iterator, Literal, Mapping, Optional, Sequence, Tuple, cast
 
 from dagster import AssetKey
 from dagster._annotations import experimental, public
+from sqlglot import exp, parse_one, to_table
+from sqlglot.optimizer import Scope, build_scope, optimize
+
+LookMLStructureType = Literal["dashboard", "explore", "table", "view"]
+
+logger = logging.getLogger("dagster_looker")
+
+
+def build_deps_for_looker_dashboard(
+    dagster_looker_translator: "DagsterLookerTranslator",
+    lookml_structure: Tuple[Path, LookMLStructureType, Mapping[str, Any]],
+) -> Sequence[AssetKey]:
+    lookml_view_path, _, lookml_dashboard_props = lookml_structure
+
+    return list(
+        {
+            dagster_looker_translator.get_asset_key(
+                lookml_structure=(
+                    lookml_view_path,
+                    "explore",
+                    lookml_element_from_dashboard_props,
+                )
+            )
+            for lookml_element_from_dashboard_props in itertools.chain(
+                # https://cloud.google.com/looker/docs/reference/param-lookml-dashboard#elements_2
+                lookml_dashboard_props.get("elements", []),
+                # https://cloud.google.com/looker/docs/reference/param-lookml-dashboard#filters
+                lookml_dashboard_props.get("filters", []),
+            )
+            if lookml_element_from_dashboard_props.get("explore")
+        }
+    )
+
+
+def build_deps_for_looker_explore(
```

Comment:
Do we want these exposed either as top-level exports or as methods on the translator so that users can call them themselves? e.g. a user wants to call `build_deps_for_looker_explore` & then modify the result rather than rebuilding logic themselves

Reply:
We can consider exposing them to the top-level if there's a need for it. I want to see how users react first before we push stuff up as top-level imports.

Worse comes to worst, they can just import from the absolute path if they need it asap